internal/pkg/v2/infrastructure/redis/event.go

```diff
@@ -25,6 +29,84 @@ const (
 	EventsCollectionReadings   = EventsCollection + ":readings"
 )
 
+// deleteEventsByIdChannel channel used to delete events asynchronously
+var DeleteEventsByIdsChannel = make(chan []string, 50)
+
+// AsyncDeleteEvents Handles the deletion of device events asynchronously. This function is expected to be running
+// in a go-routine and works with the "DeleteEventsByIds" function for better performance.
+func AsyncDeleteEvents(dic *di.Container) {
+	lc := container.LoggingClientFrom(dic.Get)
+	lc.Debug("Starting background event deletion process")
+	dbClient := v2DataContainer.DBClientFrom(dic.Get)
+	for {
+		select {
+		case eventIds, ok := <-DeleteEventsByIdsChannel:
+			if ok {
+				lc.Debug(fmt.Sprintf("Prepare to delete %v events", len(eventIds)))
+				startTime := time.Now()
+				err := dbClient.DeleteEventsByIds(eventIds)
+				if err != nil {
+					lc.Error(fmt.Sprintf("Deleted events failed.  Err: %s", err.Error()))
+				} else {
+					lc.Debug(fmt.Sprintf("Deleted events successfully. elapsed time: %s", time.Since(startTime)))
+				}
+			}
+		}
+	}
+}
+
+// DeleteEventsByIds deletes all events with given event Ids
+func (c *Client) DeleteEventsByIds(eventIds []string) (edgeXerr errors.EdgeX) {
```

Comment:
@cloudxxx8 , @judehung , After more consideration I think we can do away with the async channels. All they do is provide background processing of the deletes. This can be done in a much simpler way here if we have deletePushedEvents() (like others funcs in this layer) that uses a short running Go func (i.e. returns after complete). 

Here is some pseudo code
```
func deletePushedEvents(conn redis.Conn) edgeXerr errors.EdgeX  {
    get the Ids for events that have been pushed
   if err then 
       return error

   go func(eventIds string) {
       get each event by id, remove the Event's readings and then remove the Event 
       log any errors, but errors don't stop the deleting of other events or readings.
   }
}
```